include/libtorrent/span.hpp

```diff
@@ -60,10 +60,12 @@ namespace libtorrent
 		span(U (&arr)[N])
 			: m_ptr(&arr[0]), m_len(N) {}
 
+#if (defined __GNUC__ || (defined _MSC_VER && _MSC_VER >= 1900))
```

Comment:
Yes, it will fail, but in general, it will fail anyway if msvc < 14 and gcc/clang is not modern enough